planner/core/memtable_predicate_extractor.go

```diff
@@ -640,3 +640,58 @@ func (e *InspectionSummaryTableExtractor) Extract(
 	e.MetricNames = metricNames
 	return remained
 }
+
+// SlowQueryExtractor is used to extract some predicates of `slow_query`
+type SlowQueryExtractor struct {
+	extractHelper
+
+	SkipRequest bool
+	StartTime   time.Time
+	EndTime     time.Time
+	Enable      bool
+}
+
+// Extract implements the MemTablePredicateExtractor Extract interface
+func (e *SlowQueryExtractor) Extract(
+	ctx sessionctx.Context,
+	schema *expression.Schema,
+	names []*types.FieldName,
+	predicates []expression.Expression,
+) []expression.Expression {
+	remained, startTime, endTime := e.extractTimeRange(ctx, schema, names, predicates, "time", ctx.GetSessionVars().StmtCtx.TimeZone)
+	e.setTimeRange(startTime, endTime)
+	e.SkipRequest = e.Enable && e.StartTime.After(e.EndTime)
+	if e.SkipRequest {
+		return nil
+	}
+	return remained
+}
+
+func (e *SlowQueryExtractor) setTimeRange(start, end int64) {
```

Comment:
This function has the same logic with `MetricTableExtractor::getTimeRange` see: https://github.com/pingcap/tidb/blob/f37cfef1993d1d96c0e2b494711daac5d2d47acb/planner/core/memtable_predicate_extractor.go#L554

Reply:
No, when user doesn't specify the time, the `MetricTableExtractor ` will set time to last `10min`, but `SlowQueryExtractor ` will do nothing, Then the executor will parse the current slow-log file,  make this behavior compatible with before.